src/Channels.Networking.Sockets/SocketConnection.cs

```diff
@@ -0,0 +1,304 @@
+﻿using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Channels.Networking.Sockets
+{
+    public class SocketConnection : IChannel
+    {
+        private Socket _socket;
+        private Channel _input, _output;
+        private ChannelFactory _channelFactory;
+        private readonly bool _ownsChannelFactory;
+
+        static readonly EventHandler<SocketAsyncEventArgs> _asyncCompleted = OnAsyncCompleted;
+
+        public IReadableChannel Input => _input ?? CreateInput();
```

Comment:
Fixed:

                // wait for someone to be interested in data before we
                // start allocating buffers and probing the socket
                await _input.ReadingStarted;